emissionsapi/db.py

```diff
@@ -41,6 +42,50 @@ class File(Base):
     """Name of processed data file"""
 
 
+class Cache(Base):
+    """ORM object for the request cache
+    """
+    __tablename__ = 'cache'
+    request = Column(String, primary_key=True)
+    """Primary key identifying the request"""
+    begin = Column(DateTime)
+    """Begin of the time interval involved in this request (used for
+    efficiently invalidating caches)
+    """
+    end = Column(DateTime)
+    """End of the time interval involved in this request (used for efficiently
+    invalidating caches)
+    """
+    response = Column(PickleType)
+    """Cached response"""
+
+    @classmethod
+    def invalidate(cache, session, begin, end):
+        """Invalidates/deletes all cached responses in the given interval to
+        ensure these data is generated anew. This is meant to be run when the
+        underlying data for this interval changes, for instance since new data
+        has been imported.
+
+        Cached responses without well defined interval – begin or end is NULL –
+        will always be invalidated by this method.
+        :param session: SQLAlchemy Session
+        :type session: sqlalchemy.orm.session.Session
+        :param begin: Begin of the interval to invalidate
+        :type begin: datetime.datetime
+        :param end: End of the interval to invalidate
+        :type end: datetime.datetime
+
+        """
+        logger.debug('Invalidating cache in interval %s..%s',
+                     begin.isoformat(), end.isoformat())
+        session.query(cache)\
+               .filter(or_(cache.begin.is_(None),
+                           cache.end.is_(None),
+                           and_(cache.begin < end,
+                                cache.end >= begin)))\
```

Comment:
You can also not invalidate when only `cache.begin` or `cache.end` is given and the new points are before or after the cache one.